### ISSUE TITLE
../aten/src/ATen/native/cuda/Loss.cu:92: operator(): block: [37,0,0], thread: [96,0,0] Assertion `input_val >= zero && input_val <= one` failed.

### ISSUE THREAD

**Issue by ramonhollands** [LSH9832/edgeyolo#16]
Hi Shihan Liu,

Thanks for your work on this repo!

I'm trying to run a custom training myself, using your train script and the yolo formats. The dataset seems to be fine, looking at demo/load_dataset.py. Any ideas where I can start my bug hunt, given above error?

Best regards,
Ramon

**Comment by LSH9832** [comment]
this error occurs when
1. there's at least one NAN(not a number) element in your prediction, check by using torch.isnan(tensor). if so, reducing your learning rate might solve your problem, but sometimes it is because your installed CUDA version is too high for your GPU device(s)
```python
for name, param in self.model.named_parameters():
    if param.grad is not None and torch.isnan(param.grad).any():
        print("nan gradient name:", name)
```
2. the length of prediction does not match the length of labels, please check your dataset config and make sure it is correct.

**Comment by ramonhollands** [comment]
Thanks for your fast reply.

It looks like it's training correctly after changing the fp16 setting to false again. Could that be the cause? First 30 epochs have succeeded so far. I'll keep you posted.

**Comment by LSH9832** [comment]
what type is your GPU? I guess that your GPU might not fully support AMP(auto-mixed-precision) training, I test training with fp16 enabled and it works well in RTX30 sesries and Tesla T4

**Comment by ramonhollands** [comment]
It's a RTX3090, running CUDA 11.4. I am doing some experiments these days and let you know. If you are interested in this dataset, I can share it with you.

Looking forward to eventually contribute to this repo.

**Comment by yydc-0** [comment]
> of the CUDA

I have the same problem.

**Comment by yydc-0** [comment]
> Thanks for your fast reply.
> 
> It looks like it's training correctly after changing the fp16 setting to false again. Could that be the cause? First 30 epochs have succeeded so far. I'll keep you posted.

I have the same problem after seting the fp16 to true.
